parser/ini.go

```diff
@@ -0,0 +1,128 @@
+package parser
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type IniData map[string]map[string]string
+
+type Parser struct {
+	data IniData
+}
+
+func NewParser() *Parser {
+	parser := Parser{}
+	parser.data = make(IniData)
+	return &parser
+}
+func (parser *Parser) readFromReader(r io.Reader) error {
+	var section string
+
+	sc := bufio.NewScanner(r)
+
+	for sc.Scan() {
+
+		line := strings.Trim(sc.Text(), " \n\t")
+		TrimmedLine := strings.TrimSpace(line)
+		if len(line) > 2 && line[0] == '[' && line[len(line)-1] == ']' {
+
+			section = strings.TrimSpace(line[1 : len(line)-1])
+			parser.data[section] = make(map[string]string)
+
+		} else if strings.Contains(line, "=") {
+
+			keyVal := strings.Split(line, "=")
+			key := strings.TrimSpace(keyVal[0])
+			val := strings.TrimSpace(keyVal[1])
+			parser.data[section][key] = val
+
+		} else if len(TrimmedLine) == 0 {
+			continue
+		} else if strings.Contains(TrimmedLine, "#") {
+			continue
+		} else {
+			return fmt.Errorf("No sections found")
+		}
+
+	}
+	return nil
+}
+func (parser *Parser) ReadFromString(content string) error {
+	return parser.readFromReader(strings.NewReader(content))
+
+}
+
+func (parser *Parser) ReadFromFile(path string) error {
+	fh, err := os.Open(path)
+
+	if err != nil {
+		return fmt.Errorf("Could not open file '%v': %v", path, err)
+	}
+	return parser.readFromReader(fh)
+
+}
+
+func (parser *Parser) Get(section, key string) (string, error) {
+	if parser.data[section][key] == "" {
```

Comment:
the check here is problematic because you can have an empty value 
```
[User]
name = Ahmed
Password = 
```
here the key is password and its value is `""`  you can directly do return parser.data[section][key] and it already returns string, error `plain go map data access`